blockly/apps/blocklyduino/blockly_helper.js

```diff
@@ -38,14 +38,16 @@ function restore_blocks() {
 function save() {
   var xml = Blockly.Xml.workspaceToDom(Blockly.mainWorkspace);
   var data = Blockly.Xml.domToText(xml);
-
+  var fileName = window.prompt("What would you like to name your file?", "BlocklyDuino");
+  fileName = fileName + ".xml";
   // Store data in blob.
   // var builder = new BlobBuilder();
   // builder.append(data);
   // saveAs(builder.getBlob('text/plain;charset=utf-8'), 'blockduino.xml');
-  console.log("saving blob");
-  var blob = new Blob([data], {type: 'text/xml'});
-  saveAs(blob, 'blockduino.xml');
+  if(fileName != "null"){
```

Comment:
This will never result in a false state as you are adding ".xml" on line 42. Not quite sure why checking for the string null though instead of just `null`.

I'd suggest to remove line 42, then `if (fileName !== null)` here and in line 49 change to `saveAs(blob, fileName + '.xml')`